controllers/UserManagement/UserCreateController.js

```diff
@@ -0,0 +1,103 @@
+const BaseController = require('../BaseController');
+const {orgLookup} = require('../../services/UserManagement/OrganisationSearchService');
+const { postRequest } = require('../../services/UserManagement/UserActionsService');
+
+class UserCreateController extends BaseController {
+
+  async getValues(req, res, next) {
+
+    try {
+      const orgTeamsResult = await orgLookup({
+        ...this.getOptions(req),
+        url: `/admin/organisations/${req.params.orgId}/teams`
+      });
+
+      req.sessionModel.set('orgTeamsToSelect', orgTeamsResult);
+      req.sessionModel.set('orgId', req.params.orgId);
+      if (req.query['currentTeamId']) {
+        req.sessionModel.set('currentTeamId', req.query['currentTeamId']);
+      }
+
+      next();
+
+    } catch (err) {
+      err.template = 'errors/organisation-error';
+      next(err);
+    }
+  }
+
+  async process(req, res, next) {
+    // using it for pre-populating fields and remain the user entered values on the error screen
+    req.sessionModel.set('firstNameUserEntered', req.form.values['firstName']);
+    req.sessionModel.set('lastNameUserEntered', req.form.values['lastName']);
+    req.sessionModel.set('emailUserEntered', req.form.values['email']);
+    req.sessionModel.set('teamIdUserSelected', req.form.values['teamSelect']);
+
+    next();
+  }
+
+  async saveValues(req, res) {
+
+    // Get the values from the edit form
+    const firstname = req.form.values['firstName'];
+    const lastname = req.form.values['lastName'];
+    const email = req.form.values['email'];
+    const  teamIdToAdd = req.form.values['teamSelect'];
+
+    try {
+      req.sessionModel.set('addedUser', true);
+      await postRequest( {
+        ...this.getOptions(req),
+        url: `/admin/organisations/${req.params.orgId}/users`,
+      }, { firstname: firstname, lastname: lastname, email: email, teamId: teamIdToAdd });
+
+      req.sessionModel.set('userFullName', firstname + ' ' + lastname);
+      req.sessionModel.unset('firstNameUserEntered');
+      req.sessionModel.unset('lastNameUserEntered');
+      req.sessionModel.unset('emailUserEntered');
+      req.sessionModel.unset('teamIdUserSelected');
+
+    } catch (err) {
+      req.sessionModel.set('userFullName', '');
```

Comment:
When I tested this, if the user already exists, it fails... which is correct. But it only fails with "There was a problem adding user". It feels as though this error should be explanatory given I can imagine this scenario occurring.

Additionally, if you enter this error state, it redirects back to the previous page, but then if you go the "Add user" page again, the form is pre-filled with the previous state. It feels like it should be cleared in this catch block

Reply:
I have updated the error message as discussed with Jo, and also cleared the pre-filled previous state on error.